src/fft_2n.h

```diff
@@ -219,23 +220,29 @@ void Radix2<T>::fft(vec::Vector<T>& output, vec::Vector<T>& input)
 template <typename T>
 void Radix2<T>::fft_inv(vec::Vector<T>& output, vec::Vector<T>& input)
 {
-    unsigned len = this->n;
+    const unsigned len = this->n;
+
+    // copy input to output
+    output.copy(&input);
+
     for (unsigned m = len / 2; m >= 1; m /= 2) {
         unsigned doubled_m = 2 * m;
         for (unsigned j = 0; j < m; ++j) {
             T r = inv_W->get(j * len / doubled_m);
             for (unsigned i = j; i < len; i += doubled_m) {
-                T a = input.get(i);
-                T b = input.get(i + m);
-                input.set(i, this->gf->add(a, b));
-                input.set(i + m, this->gf->mul(r, this->gf->sub(a, b)));
+                T a = output.get(i);
+                T b = output.get(i + m);
+                output.set(i, this->gf->add(a, b));
+                output.set(i + m, this->gf->mul(r, this->gf->sub(a, b)));
             }
         }
     }
 
-    // set output  = scramble(input), i.e. bit reversal ordering
+    // reversion of elements of output to return values on the natural order
```

Comment:
You could have a method `reverse`, it would be clearer (and avoid code duplication).